diff.py

```diff
@@ -2018,40 +2018,58 @@ def _normalize_adrp_differences(self, mnemonic: str, row: str) -> str:
         return row
 
 
-class AsmProcessorI686(AsmProcessor):
+class AsmProcessorX86(AsmProcessor):
     def process_reloc(self, row: str, prev: str) -> Tuple[str, Optional[str]]:
         if "WRTSEG" in row:  # ignore WRTSEG (watcom)
             return prev, None
         repl = row.split()[-1]
         mnemonic, args = prev.split(maxsplit=1)
         offset = False
+        addr_imm = None
 
         # Calls
+
+        # Example lcall $0x0, $0x00
+        if "lcall" in mnemonic:
+            addr_imm = re.search(r".*", args)
+
         # Example call a2f
         # Example call *0
         # Example jmp  64
-        if mnemonic in I686_BRANCH_INSTRUCTIONS:
+        elif mnemonic in X86_BRANCH_INSTRUCTIONS:
             addr_imm = re.search(r"(^|(?<=\*)|(?<=\*\%cs\:))[0-9a-f]+", args)
 
         # Direct use of reloc
-        # Example 0x0,0x8(%edi)
-        # Example 0x0,%edi
-        # Example *0x0(,%edx,4)
+        # Match 0x0 part to replace
+
         # Example %edi,0
         # Example movb $0x0,0x0
-        # Example $0x0,0x4(%edi)
-        # Match 0x0 part to replace
-        else:
+        if not addr_imm:
             addr_imm = re.search(r"(?:0x)?0+$", args)
 
+        # Example movb $0x0,0x0(%si)
+        if not addr_imm:
+            addr_imm = re.search(r"(?<=,)0x0+(?=\(.*\))", args)
+
+        # Example 0x0,0x8(%edi)
+        # Example 0x0,%edi
+        # Example *0x0(,%edx,4)
+        # Example $0x0,0x4(%edi)
         if not addr_imm:
             addr_imm = re.search(r"(^\$?|(?<=\*))0x0", args)
 
         # Offset value
+
+        # Example movb $0x0,0x4
+        # Example movb $0x0,0x4(%si)
+        if not addr_imm:
+            addr_imm = re.search(r"(?<=,)0x[0-9a-f]+", args)
+            offset = True
+
         # Example 0x4,%eax
         # Example $0x4,%eax
         if not addr_imm:
-            addr_imm = re.search(r"(^|(?<=\*)|(?<=\$))0x[0-9a-f]+", args)
+            addr_imm = re.search(r"(^|(?<=\*)|(?:\$))0x[0-9a-f]+", args)
```

Comment:
this will cause $'s to start appearing in the output, right? is that expected?

Reply:
Nope it's actually to remove the $ from the output

Reply:
Ah, I see, because we look at span.start()/.end() to know what part of the instruction to remove! I was looking at
```py
        if offset:
            repl = f"{repl}+{addr_imm.group()}"
```
which I guess still carries the `$`, but maybe that's fine. Could do an lstrip() otherwise.